File: optaplanner/src/main/java/org/example/domain/Airport.java
package org.example.domain;



import java.util.List;
import java.util.Map;


public class Airport {
    //공항 이름
    private String name;
    //공항에 따른 deadhead비용 맵 ex) deadheadCost.get("ATL") -> 200
    private Map<String, Integer> deadheadCost;

    public Airport(String name, Map<String, Integer> deadheadCost) {
        this.name = name;
        this.deadheadCost = deadheadCost;
    }

    public String getName() {
        return name;
    }

    public int getDeadheadCost(String s) {
        return deadheadCost.get(s);
    }

    @Override
    public String toString() {
        return "Airport{" +
                "name='" + name + '\'' +
                ", deadheadCost=" + deadheadCost +
                '}';
    }

    public static Airport findAirportByName(List<Airport> airports, String name) {
        for (Airport airport : airports) {
            if (airport.getName().equals(name)) {
                return airport;
            }
        }
        return null;
    }

}

File: ReinforcementLearning/Airport.py
class Airport:
    # deadheadCost = dict('name2', 'deadhead') # dictionary 자료형, name/deadhead로 인덱싱

    def __init__(self, id, name, hotelCost, destCostList):
        self.id = id
        self._name = name  # 공항 이름
        self._hotelCost = hotelCost
        self.deadheadCost = {}  # 빈 딕셔너리로 초기화
        for destCost in destCostList:
            self.deadheadCost[destCost[0]] = destCost[1]

    def getDeadheadCost(self, dest):
        return self.deadheadCost[dest.name]

    @property
    def name(self):
        return self._name

    @name.setter
    def name(self, value):
        self._name = value

    @property
    def hotelCost(self):
        return self._hotelCost

    @hotelCost.setter
    def hotelCost(self, value):
        self._hotelCost = value

    # @deadheadCost.setter
    # def deadheadCost(self, value):
    #   self.deadheadCost = value

    # 인스턴스 이름 설정 (java에서 toString())

    def __str__(self):
        return f"Airport - {self.name}"

    def find_airport(self, airports, name):
        airport = next(
            (airport for airport in airports if airport.name == name), None)
        if airport is None:
            raise ValueError("Airport not found")
        return airport
